ui/src/components/uploader/QasUploader.vue

```diff
@@ -87,56 +86,77 @@ export default {
       type: String
     },
 
-    maxFiles: {
-      default: 1,
-      type: Number
-    },
-
     value: {
       default: '',
       type: [Array, String]
-    }
+    },
+
+    errorMessage: {
+      type: String,
+      default: ''
+    },
+
+    error: {
+      type: Boolean
+    },
+
+    readonly: {
+      type: Boolean
+    },
 
+    maxFiles: {
+      type: Number,
+      default: undefined
+    }
   },
 
   data () {
     return {
-      files: [],
       isFetching: false,
-      paths: {}
+      isUploading: false,
+      isImage: true
     }
   },
 
   computed: {
-    hasAPIValue () {
-      return this.value.startsWith('https://s3.amazonaws.com/')
+    self () {
+      return this
     },
 
-    hasBottom () {
-      return !!this.error || !!this.errorMessage || !!this.hint
+    uploaderClass () {
+      return this.hasCustomUpload ? 'hidden' : 'fit'
     },
 
-    imageName () {
-      return `${this.value}`.split('/').pop()
+    pathsLoop () {
+      return Array.isArray(this.value) ? this.value : (this.value ? [this.value] : [])
     },
 
-    readOnly () {
-      return this.files.length >= this.maxFiles
-    }
-  },
+    showAddFile () {
+      return !this.readonly &&
+        (
+          this.maxFiles
+            ? this.value.length < this.maxFiles
+            : true
+        )
+    },
+
+    isMultiple () {
+      return this.$attrs.multiple || this.$attrs.multiple === ''
+    },
+
+    hasCustomUpload () {
```

Comment:
`return this.$slots['custom-upload'] || this.$scopedSlots['custom-upload']`

Reply:
feito.